cranelift/filetests/src/runtest_environment.rs

```diff
@@ -75,6 +103,34 @@ impl RuntestEnvironment {
 
         Ok(())
     }
+
+    /// Allocates a struct to be injected into the test.
+    pub fn runtime_struct(
+        &self,
+        mut alloc_heap: impl FnMut(u64) -> u64,
+        mut alloc_table: impl FnMut(u64, u64) -> u64,
+    ) -> Vec<u64> {
+        let context_struct = self
+            .entries
+            .iter()
+            .flat_map(|entry| match entry {
+                RuntestEntry::Heap(heap) => {
+                    let size: u64 = heap.size.into();
+                    [alloc_heap(size), size]
+                }
+                RuntestEntry::Table(table) => {
+                    let entry_size: u64 = table.entry_size.into();
+                    let entry_count: u64 = table.entry_count.into();
+                    let bytes = entry_size * entry_count;
```

Comment:
This is wrong according to `docs/ir.md`.

Table bounds should be the number of *elements* and not the number of bytes. 

However nothing works when I change it to that. Probably something to investigate further.

See:
https://github.com/bytecodealliance/wasmtime/blob/1d319c0ec2bcf5a169062eece08f7f78650ce282/cranelift/docs/ir.md?plain=1#L769-L770